x/mint/keeper/mint.go

```diff
@@ -0,0 +1,53 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/dymensionxyz/dymension-rdk/x/mint/types"
+)
+
+func (k Keeper) HandleMintingEpoch(ctx sdk.Context) (sdk.Coins, error) {
+	var mintedCoins sdk.Coins
+	params := k.GetParams(ctx)
+
+	//calculate coins
+	total := k.bankKeeper.GetSupply(ctx, params.MintDenom)
+	mintAmount := k.CalcMintedCoins(ctx, total.Amount)
+	if mintAmount.IsZero() {
+		return mintedCoins, nil
+	}
+
+	// mint coins, update supply
+	mintedCoins = sdk.NewCoins(sdk.NewCoin(params.MintDenom, mintAmount.TruncateInt()))
+	err := k.bankKeeper.MintCoins(ctx, types.ModuleName, mintedCoins)
+	if err != nil {
+		return mintedCoins, err
+	}
+
+	// send the minted coins to the fee collector account
+	err = k.DistributeMintedCoin(ctx, mintedCoins)
+	if err != nil {
+		return mintedCoins, err
+	}
+
+	return mintedCoins, nil
+}
+
+func (k Keeper) CalcMintedCoins(ctx sdk.Context, totalAmt sdk.Int) sdk.Dec {
+	params := k.GetParams(ctx)
+	minter := k.GetMinter(ctx)
+	mintAmount := minter.CurrentInflationRate.MulInt(totalAmt).QuoInt(sdk.NewInt(params.MintEpochSpreadFactor))
```

Comment:
i understand that MintEpochSpreadFactor param, is the number of minting epochs in a year that is used to calculate the minted tokens for each epoch.
If this is the case , i do not see why it is necessary to have a specific parameter for that, and why cannot be derived from MintEpochIdentifier to calculate how many minting epochs we have in a year. and in case we use it, i think MintEpochSpreadFactor needs to be validated to match the MintEpochIdentifier and probably a better name like MintingEpochsInAYear can be used.

Reply:
I'll check if possible to get it from the epochKeeper
otherwise, we need some translation between `mint_epoch - string` to `num of epochs in a year`
anyway, it allow some generalized approach, in case inflation won't be calculated annually